### ISSUE TITLE
ActivityEventAttachingLogProcessor - StateConverter is non-functional

### ISSUE THREAD

**Issue by Jaskaranbir** [open-telemetry/opentelemetry-dotnet-contrib#754]
# Issue with OpenTelemetry.Extensions

Versions used:

```
OpenTelemetry - 1.4.0-beta.2
OpenTelemetry.Api - 1.4.0-beta.2
OpenTelemetry.Exporter.OpenTelemetryProtocol - 1.4.0-beta.2
OpenTelemetry.Exporter.OpenTelemetryProtocol.Logs - 1.4.0-beta.2
OpenTelemetry.Extensions.Hosting - 1.0.0-rc9.8
OpenTelemetry.Instrumentation.AspNetCore - 1.0.0-rc9.8
OpenTelemetry.Instrumentation.Http - 1.0.0-rc9.8
```

Runtime version: 6.0

**Is this a feature request or a bug?**

* [ ] Feature Request
* [x] Bug

**What is the expected behavior?**

The [DefaultLogStateConverter](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Extensions/Internal/DefaultLogStateConverter.cs) should correctly pull data from `LogRecord` and set it as tags for the ActivityEvent.

**What is the actual behavior?**

Besides adding the three tags corresponding with following lines of code:

https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Extensions/Internal/ActivityEventAttachingLogProcessor.cs#L55-L56  
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Extensions/Internal/ActivityEventAttachingLogProcessor.cs#L84

No other tags are added to the `ActivityEvent`.

## Additional Context

The current implementation assigns a `tags` object to ActivityEvent and then later on mutates this object with additional properties:

https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Extensions/Internal/ActivityEventAttachingLogProcessor.cs#L53  
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Extensions/Internal/ActivityEventAttachingLogProcessor.cs#L72  
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Extensions/Internal/DefaultLogStateConverter.cs#L39

However, the `ActivityEvent` implementation creates its own copy of tags passed via constructor, so any mutations beyond that will not be registered:

https://github.com/dotnet/runtime/blob/4ad19b10efabbbf8de5a38185bf3402661e01a5a/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivityEvent.cs#L36  
https://github.com/dotnet/runtime/blob/4ad19b10efabbbf8de5a38185bf3402661e01a5a/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L1546-L1569

So regardless of whether the default `StateConverter` used or custom implementation, it'll never work. So the current code for this should be changed to return the tags instead, and append them to pre-existing tags in `ActivityEventAttachingLogProcessor` than relying on mutations.

**Comment by TimothyMothra** [comment]
Looks like this was fixed already as part of #680

**Comment by Jaskaranbir** [comment]
Yes, this is resolved in the newer version.
